cse/CommentSpider.py

```diff
@@ -1,45 +1,53 @@
+import os
+import re
 import scrapy
-from scrapy.crawler import CrawlerProcess
 from scrapy.selector import Selector
-import re
-import json
-from pprint import pprint
-import hashlib
+from scrapy import signals
 
 from cse.WpApiDataPipelineBootstrap import WpApiDataPipelineBootstrap as PipelineBootstrap
 from cse.CSVWriter import CSVWriter
 
 class CommentSpider(scrapy.Spider):
     # this spider scrapes a single article within the domain washingtonpost.com (https://www.washingtonpost.com/)
     name = 'washingtonpost.com'
-    urls = [['https://www.washingtonpost.com/politics/courts_law/supreme-court-to-consider-major-digital-privacy-case-on-microsoft-email-storage/2017/10/16/b1e74936-b278-11e7-be94-fabb0f1e9ffb_story.html'],
-            ['https://www.washingtonpost.com/politics/japanese-leader-shinzo-abe-plays-the-role-of-trumps-loyal-sidekick/2017/11/06/cc23dcae-c2f1-11e7-afe9-4f60b5a6c4a0_story.html'],
-            ['http://www.washingtonpost.com/news/the-fix/wp/2017/11/03/trumps-latest-target-the-diversity-visa-program-wasnt-always-aimed-at-achieving-diversity/']
+    urls = [
+        ['https://www.washingtonpost.com/politics/courts_law/supreme-court-to-consider-major-digital-privacy-case-on-microsoft-email-storage/2017/10/16/b1e74936-b278-11e7-be94-fabb0f1e9ffb_story.html'],
+        ['https://www.washingtonpost.com/politics/japanese-leader-shinzo-abe-plays-the-role-of-trumps-loyal-sidekick/2017/11/06/cc23dcae-c2f1-11e7-afe9-4f60b5a6c4a0_story.html'],
+        ['http://www.washingtonpost.com/news/the-fix/wp/2017/11/03/trumps-latest-target-the-diversity-visa-program-wasnt-always-aimed-at-achieving-diversity/']
     ]
 
     __pbs = None
+    __writer = None
 
 
     def __init__(self):
         self.__pbs = PipelineBootstrap()
         self.__pbs.setupPipeline()
+        self.__setupFileWriter("comments.csv")
+
 
+    @classmethod
```

Comment:
where is this method used?

Reply:
it is called by scrapy itself to register the signal (shutdown hook / teardown hook)

I use this hook to close the file handle after processing all urls.